components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -2156,4 +2104,32 @@ private String getTenantUrl(String url, String tenantDomain) throws URISyntaxExc
         return uriModified.toString();
     }
 
+    private void verifyAndUpdateRoleConfiguration(String tenantDomain, int tenantId,
+            PermissionsAndRoleConfig roleConfiguration) throws IdentityProviderManagementException {
+
+        List<RoleMapping> validRoleMappings = new ArrayList<>();
+        List<String> validIdPRoles = new ArrayList<>();
+
+        for (RoleMapping mapping : roleConfiguration.getRoleMappings()) {
+            try {
+                UserStoreManager usm = IdPManagementServiceComponent.getRealmService().getTenantUserRealm(tenantId)
+                        .getUserStoreManager();
+                String role = mapping.getLocalRole().getLocalRoleName();
+                if (mapping.getLocalRole().getUserStoreId() != null) {
+                    role = mapping.getLocalRole().getUserStoreId() + CarbonConstants.DOMAIN_SEPARATOR + role;
+                }
+                // Remove invalid mappings if local role does not exists.
+                if (usm.isExistingRole(role)) {
+                    validRoleMappings.add(mapping);
```

Comment:
I guess better to have debug level logs here..?

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/2155